packages/cfd/src/Components/cfd-real-account-display.tsx

```diff
@@ -162,9 +160,7 @@ const CFDRealAccountDisplay = ({
     }, [residence, is_logged_in, is_eu, is_eu_country, platform]);
 
     const onSelectRealSynthetic = () => {
-        if (is_eu && standpoint.malta && !has_malta_account) {
-            openAccountNeededModal('malta', localize('Deriv Synthetic'), localize('DMT5 Synthetic'));
```

Comment:
This code is unreachable because this function will never be called for a EU client with standpoint.malta=true.
This function is normally called when a client clicks an 'Add real account' button (this button is called 'Select' when has_cfd_account=true) on a Synthetic card or an '+Add region' button that is shown under Synthetic card for non-EU clients only.
We show Synthetic account card to EU clients if they have an mt5.real.synthetic account created already (as per isSyntheticCardVisible logics at line 314 in cfd-dashboard.tsx). Otherwise, they can see 'CFDs card only.
Also, if an account is already added, we don't show 'Add real account'('Select') button on the account card.
Hypothetically, if we decided to show Synthetic account card to EU clients when they don’t have a created mt5.real.synthetic account yet, we would anyway disable ‘Add real account’/‘Select’ button on the Synthetic card for clients who have standpoint.malta=true (isMT5AccountCardDisabled would return true line 212 for this case).